### PR TITLE
Fixing Public Jira Issue : APIMANAGER-1821

### PR DESCRIPTION
**Pull request by ChamNDeSilva**
Adding token validation time to authentication API request logs

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIAuthenticationHandler.java [hunk 1]
```diff
@@ -103,6 +103,13 @@ private void initializeAuthenticator() {
     }
 
     public boolean handleRequest(MessageContext messageContext) {
+        long startTime = System.nanoTime();
+        long endTime;
+        long difference;
+
+        endTime = System.nanoTime();
+        difference = (endTime - startTime) / 1000000;
```
Comment (nuwand) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIAuthenticationHandler.java]: End time should be calculated after the authentication has happened.

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIAuthenticationHandler.java [hunk 2]
```diff
@@ -103,6 +103,13 @@ private void initializeAuthenticator() {
...
+        String messageDetails = logMessageDetails(messageContext, difference);
```
Comment (nuwand) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIAuthenticationHandler.java]: Should only call this method if debug log is enabled.

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIAuthenticationHandler.java [hunk 3]
```diff
@@ -227,6 +238,10 @@ private void logMessageDetails(MessageContext messageContext) {
         if (userAgent != null) {
             logMessage = logMessage + " with userAgent=" + userAgent;
         }
+        String accessToken = (String) ((TreeMap) axisMC.getProperty("TRANSPORT_HEADERS")).get("Authorization");
```
Comment (nuwand) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIAuthenticationHandler.java]: Use Constants.

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIAuthenticationHandler.java (hunk 1 above)
Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIAuthenticationHandler.java]: Fixed in PR https://github.com/wso2/carbon-apimgt/pull/1073

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIAuthenticationHandler.java (hunk 2 above)
Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIAuthenticationHandler.java]: Fixed #1073

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIAuthenticationHandler.java (hunk 3 above)
Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIAuthenticationHandler.java]: Fixed in https://github.com/wso2/carbon-apimgt/pull/1041

**Pull request merged**